### PR TITLE
[XLA:GPU][ROCm] adding a DISABLED_ON_GPU_ROCM macro 

### PR DESCRIPTION
**Pull request by whchung**
adding a DISABLED_ON_GPU_ROCM macro to disable subtests that are not yet supported on ROCm.

Applying that macro to a few subtests in convolution_test.cc and convolution_variants_test.cc

### CODE REVIEW

File: tensorflow/compiler/xla/tests/convolution_test.cc [hunk 1]
```diff
@@ -1946,7 +1948,9 @@ XLA_TEST_F(ConvolutionTest, ConvolveF32BackwardInputGroupedConvolution) {
 
 class ConvolutionHloTest : public HloTestBase {};
 
-XLA_TEST_F(ConvolutionHloTest, ConvolveF64Forward) {
+// double datatype is not yet supported in ROCm
+XLA_TEST_F(ConvolutionHloTest,
+           DISABLED_ON_GPU_ROCM(DISABLED_ON_CPU(ConvolveF64Forward))) {
```
Comment (thomasjoerg) [inline tensorflow/compiler/xla/tests/convolution_test.cc]: Why did you add DISABLED_ON_CPU here and below?

File: tensorflow/compiler/xla/tests/convolution_test.cc [hunk 2]
```diff
@@ -408,7 +408,9 @@ class Convolve1D_1x2x5_1x2x2_WithPadding : public ConvolutionTest {
 TYPED_TEST_CASE(Convolve1D_1x2x5_1x2x2_WithPadding, TestTypes);
 TYPED_TEST(Convolve1D_1x2x5_1x2x2_WithPadding, Types) { this->RunTest(); }
 
-XLA_TEST_F(ConvolutionTest, Convolve3D_1x4x2x3x3_2x2x2x3x3_Valid) {
+// 5D tensors are not yet supported in ROCm
+XLA_TEST_F(ConvolutionTest,
+           DISABLED_ON_GPU_ROCM(Convolve3D_1x4x2x3x3_2x2x2x3x3_Valid)) {
```
Comment (thomasjoerg) [inline tensorflow/compiler/xla/tests/convolution_test.cc]: Please add a comment why these tests are not good for ROCm at present. Bonus points for linking to a Github issue ;-)

**Review by thomasjoerg — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/xla/tests/convolution_test.cc (hunk 1 above)
Comment (whchung) [reply tensorflow/compiler/xla/tests/convolution_test.cc]: thanks for catching this. let me fix it.

File: tensorflow/compiler/xla/tests/convolution_test.cc (hunk 2 above)
Comment (whchung) [reply tensorflow/compiler/xla/tests/convolution_test.cc]: will do.

Comment (whchung) [reply tensorflow/compiler/xla/tests/convolution_test.cc]: after reviewing with @deven-amd and @jerryyin we actually already have 5D tensor / 3D convolution support in ROCm. reverting changes here.

**Comment by whchung** [comment]
@thomasjoerg Thanks for the review. After checking with @deven-amd and @jerryyin I've revised this PR to reflect the latest status on ROCm, where 3D convolution on 5D tensors are actually supported.

**Comment by whchung** [comment]
@thomasjoerg a gentle ping.

**Review by thomasjoerg — APPROVED** [review]
